src/Utilidades/Gramatica/ReglaProduccion.java

```diff
@@ -47,7 +49,10 @@ public boolean getMarcadoSiguiente(){
 
     //setters
     public void setSimboloGramatical(String simboloGramatical){
-        this.simboloGramatical = simboloGramatical;
+        SimboloGramatical simboloAux = new SimboloGramatical(simboloGramatical);
+
+
+        this.simboloGramatical = new SimboloGramatical(simboloAux);
```

Comment:
Puedes ahorrar este paso de la muerte tan sólo poniendo: this.simboloGramatical = new SimboloGramatical(simboloGramatical)

Reply:
xd